sql/index/pilosalib/mapping.go

```diff
@@ -35,21 +35,35 @@ func newMapping(path string) *mapping {
 	return &mapping{path: path}
 }
 
-func (m *mapping) open() {
-	m.openCreate(false)
+func (m *mapping) open() error {
+	return m.openCreate(false)
 }
 
 // openCreate opens and sets creation mode in the database.
-func (m *mapping) openCreate(create bool) {
+func (m *mapping) openCreate(create bool) error {
 	m.clientMut.Lock()
 	defer m.clientMut.Unlock()
+	m.mut.Lock()
+	defer m.mut.Unlock()
+
+	if m.clients == 0 && m.db == nil {
```

Comment:
so maybe instead of having here 2 mutexes we can make `m.clients` a `sync.atomic` var. WDYT?

Reply:
that other mutex it's protecting `created` as well